File: src/main/java/org/labun/jooq/generator/util/TypeResolver.java
package org.labun.jooq.generator.util;

import org.jooq.util.ColumnDefinition;
import org.labun.jooq.generator.task.TaskContext;

/**
 * @author Konstantin Labun
 */
public interface TypeResolver {
    String resolve(TaskContext context, ColumnDefinition column);
}

File: src/main/kotlin/org/labun/jooq/generator/util/TypeResolver.kt
package org.labun.jooq.generator.util

import org.jooq.util.ColumnDefinition
import org.jooq.util.DataTypeDefinition
import org.labun.jooq.generator.task.SubGeneratorContext

/**
 * @author Konstantin Labun
 */
interface TypeResolver {

  fun resolve(context: SubGeneratorContext, def: DataTypeDefinition): String
}
